### PR TITLE
[Model] Support `pacific_i64` which uses Token routed i64 in MoE

### PR DESCRIPTION
**Pull request by Complexity-ML**
## Purpose

Add support for the **Pacific-I64** model (`DeepForCausalLM`), a 170M parameter decoder-only transformer with three architectural innovations:

- **INL Dynamics**: PID-like control with velocity tracking (alpha, beta, gate, mu) for numerical stability during token generation
- **Token-Routed I64 MLP**: Deterministic integer expert routing (`token_id % num_experts`) — pure i64 control flow, no softmax gating
- **Mu-Guided Attention**: Top-down influence from previous layer's equilibrium state on Q/K/V projections

Paper under review at **TMLR** ([OpenReview](https://openreview.net/forum?id=jZq6EVboC6&noteId=jZq6EVboC6)).
Checkpoint: [Pacific-Prime/pacific-tiny-chat](https://huggingface.co/Pacific-Prime/pacific-tiny-chat)

### Why Token-Routed I64 is a separate layer (not fused_moe)

Standard `fused_moe` assumes float-based gating: `logits → softmax → top-k → float weights → scatter/gather`. Token-Routed I64 uses **pure integer routing** (`token_id % num_experts`) with no softmax, no float weights, and no top-k selection. The routing is deterministic and data-independent, which is fundamentally incompatible with fused_moe's gating interface. The layer is self-contained under `model_executor/layers/token_routed_i64/` and does not modify any existing vLLM layers.

### Also fixes two bugs in piecewise CUDA graph compilation

1. **Cycle in `split_graph()`**: `getitem` nodes on splitting_op outputs were assigned to the wrong subgraph, causing cycle detection failure in `torch.fx.passes.split_module`
2. **Data corruption during CUDA graph replay**: splitting_ops that allocate new output tensors (e.g. via `torch.bmm`) caused stale addresses in the next compiled piece's CUDA graph

## Test Plan

```bash
python -m vllm.entrypoints.openai.api_server \
    --model Pacific-Prime/pacific-tiny-chat \
    --dtype float16

### CODE REVIEW

**Review by hmellor — CHANGES_REQUESTED** [review]
- No explanation in the PR description
- i64 appears to be:
  - an unproven research experiment which is used only by this model
  - Unreleased/closed source (it was apparently used in https://huggingface.co/Pacific-Prime/pacific-prime, but this has been deleted since they announced it in January)
- Duplicates the structure of `layers/fused_moe` rather than integrating with it. It's fine for models to use layers which aren't in `model_executor/layers` but they should not be added to `model_executor/layers` if they are specific to 1 architecture
